src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp

```diff
@@ -348,6 +348,15 @@ PerformanceImplicitGemmForwardV4R4Xdlops_Padded_Gemm::CalculateGemmABlockCopyPer
         // GemmKPack is src vector read dimension, bounded by GemmKPack
         SrcDataPerRead_GemmKPack = gcd(SrcDataPerRead_GemmKPack, GemmKPack);
 
+        int gemm_k_extra = 0;
+
+        std::tie(std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, gemm_k_extra) =
+            ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm::CalculateGemmSize(
+                ctx, GemmMFactor, GemmNFactor, GemmKFactor);
+
+        if (gemm_k_extra != 0)
+            SrcDataPerRead_GemmKPack = gcd(SrcDataPerRead_GemmKPack, gemm_k_extra);
```

Comment:
should be:
```SrcDataPerRead_GemmKPack = gcd(SrcDataPerRead_GemmKPack, gemm_k_real);```

because you want to make sure ```SrcDataPerRead_GemmKPack```is a factor or ```gemm_k_real``` not ```gemm_k_extra```

(I don't know what you call ```gemm_k_real``` in your code though, haven't read all the code)